src/nidcpower/templates/session.py/delegate_to_private_method.py.mako

```diff
@@ -0,0 +1,13 @@
+<%page args="f, config, method_template"/>\
+<%
+    import build.helper as helper
+
+    suffix = method_template['method_python_name_suffix']
+%>\
+    def ${f['python_name']}${suffix}(${helper.get_params_snippet(f, helper.ParameterUsageOptions.SESSION_METHOD_DECLARATION)}):
+        '''${f['python_name']}
+
+        ${helper.get_function_docstring(f, False, config, indent=8)}
+        '''
+        return self._${f['python_name']}${suffix}(${helper.get_params_snippet(f, helper.ParameterUsageOptions.SESSION_METHOD_PASSTHROUGH_CALL)})
```

Comment:
Here there is a hidden contract that the private method's name will be identical to the public's method name; even though there's nothing enforcing that elsewhere. The method names can be specified to be arbitrary via metadata.

There is another less evil hidden contract that the parameter names are the same. This I think is more defensible.

I like that you are trying to solve this in a generic manner (but notice the template is specific to nidcpower based on its location) but I don't know that it's worth this design compromise. Are there other opportunities for us to simplify nimi-python and use this template? If there isn't I think we're better off not prematurely generalizing and making this template specific to `get_channel_names()`.

Now, there is a need to do the same in `nidigital` per #1758 so what I think you need to do is the following:

1. Create a `get_channel_names` template. Don't make it `nidcpower`-specific. This should be added and unit-tested in `nifake`.
2. Apply it to `nidcpower`
3. Apply it to `nidigital`

In separate PRs.

Reply:
> Are there other opportunities for us to simplify nimi-python and use this template?

I'm not aware of any currently.

> Create a get_channel_names template. Don't make it nidcpower-specific. This should be added and unit-tested in nifake.

I've created #1760 for this.

> Apply it to nidcpower

I'll apply the change from #1760 to this PR once it's submitted.

> Apply it to nidigital

I've updated the description of #1758 to also address this comment when that issue is being worked on.